### PR TITLE
Add environmental noise meter to calibration screen

### PR DESCRIPTION
**Pull request by IsmaelMartinez**
## Summary

Adds an optional microphone-based ambient noise meter to the calibration screen that warns users when background noise exceeds recommended levels for accurate hearing testing.

### Changes

- **New module `src/audio/noise-meter.ts`**: Core noise metering functionality
  - `rmsToDbSpl()`: Converts time-domain RMS values to approximate dB SPL using a standard consumer microphone offset (94 dB)
  - `startNoiseMeter()`: Initializes microphone access, samples audio at 200ms intervals, tracks running peak dB
  - `NOISE_WARNING_THRESHOLD_DB`: Set to 40 dB per Phase 1 spec
  - Graceful error handling for missing microphone support or denied permissions

- **Updated `src/screens/calibration.ts`**: Integrated noise meter UI
  - New "Check Ambient Noise" button that toggles noise measurement
  - Live dB readout and running peak display
  - Warning state when peak exceeds 40 dB with user-friendly messaging
  - Cleanup on navigation or test start

- **Styling in `src/styles.css`**: Visual components for the noise meter
  - `.noise-meter`: Container with tertiary background
  - `.noise-meter__readout`: Large monospace dB display
  - `.noise-meter--warning`: Warning state with amber accent

- **Comprehensive test suite `src/audio/noise-meter.test.ts`**: 225 lines covering
  - RMS-to-dB conversion accuracy and edge cases
  - Microphone initialization and permission handling
  - Sample callback invocation and peak tracking
  - Proper cleanup and idempotent stop behavior

### Design Notes

- **Non-blocking**: Users can proceed with testing regardless of noise level
- **Uncalibrated**: Consumer microphones aren't SPL-calibrated, but the 94 dB offset is a widely-used approximation that reliably flags obviously noisy environments
- **Accessibility**: Proper ARIA labels, live region updates, and screen reader announcements
- **Robustness**: Disables AGC/echo cancellation/noise suppression to get raw ambient levels; handles disconnection errors gracefully

## Related issue

Closes Phase 1 — Accuracy & Trust environmental noise check task

## Type of change

- [x] New feature
- [ ] Bug fix
- [ ] Documentation update

## Checklist

- [x] I have tested my changes (comprehensive unit test suite with 225 lines)
- [x] I have updated documentation (future-features.md updated to mark as complete)

https://claude.ai/code/session_01Gn2PHaFovXTyPMFLWDcba5

### CODE REVIEW

File: src/screens/calibration.ts [hunk 1]
```diff
@@ -93,12 +125,57 @@ export function renderCalibration(): void {
...
+    button.innerHTML = '<span aria-hidden="true">⏳</span> Requesting microphone…';
+    try {
+      noiseMeter = await startNoiseMeter((sample) => {
+        value.textContent = sample.db.toFixed(0);
+        const noisy = sample.peakDb > NOISE_WARNING_THRESHOLD_DB;
+        status.classList.toggle('noise-meter--warning', noisy);
+        message.textContent = noisy
+          ? `Peak ${sample.peakDb.toFixed(0)} dB — louder than recommended. Try a quieter room for more reliable results.`
+          : `Peak ${sample.peakDb.toFixed(0)} dB — quiet enough for testing.`;
+      });
```
[the remark this answers is not in the document]
Comment (IsmaelMartinez) [reply src/screens/calibration.ts:165]: Applied in `39f3e15` — skip redundant `textContent` writes when the rounded values haven't changed, and only `announce()` on warning-state transitions (tracked via `lastNoisy`).

File: src/audio/noise-meter.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      },
+    });
+  } catch (error) {
+    throw new AudioInitError(
+      'Microphone access was denied. Grant permission to check ambient noise.',
+      error,
+    );
+  }
+
+  const ctx = getAudioContext();
```
[the remark this answers is not in the document]
Comment (IsmaelMartinez) [reply src/audio/noise-meter.ts]: Applied in `39f3e15` — now `await ensureRunning()` so a suspended context gets resumed before the AnalyserNode starts.

File: src/screens/calibration.ts [hunk 3]
```diff
@@ -45,6 +50,27 @@ export function renderCalibration(): void {
...
+          <h2 class="card__title justify-center" id="noise-section-title"><span aria-hidden="true">🎙️</span> Environmental Noise</h2>
+          <p id="noise-description" class="text-center text-secondary">
+            Background noise can raise your thresholds. This quick check uses your microphone to estimate how quiet your room is.
+          </p>
+
+          <button class="btn btn--secondary" id="noise-check" aria-describedby="noise-description">
+            <span aria-hidden="true">🎙️</span> Check Ambient Noise
+          </button>
+
+          <div id="noise-status" class="noise-meter" role="status" aria-live="polite" hidden>
```
[the remark this answers is not in the document]
Comment (IsmaelMartinez) [reply src/screens/calibration.ts]: Applied in `39f3e15` — dropped `role="status"` / `aria-live` from the container and switched to explicit `announce()` calls triggered only when the warning state toggles (see the next thread).

File: src/audio/noise-meter.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+ * for reliable thresholds.
+ *
+ * Accuracy note: consumer microphones are not calibrated to an absolute
+ * SPL reference. The offset below is a widely-used approximation that
+ * puts typical quiet rooms around 25-35 dB and loud rooms above 60 dB,
+ * good enough to flag obviously noisy environments — not a sound-level
+ * meter replacement.
+ */
+
+import { getAudioContext, AudioInitError } from './audio-context';
```
[the remark this answers is not in the document]
Comment (IsmaelMartinez) [reply src/audio/noise-meter.ts]: Applied in `39f3e15` — swapped the import to `ensureRunning`.
